File: backend/src/main/java/com/jpromi/darts/backend/models/LocationRequest.java
package com.jpromi.darts.backend.models;

import lombok.Builder;
import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
@Builder
public class LocationRequest {
    private String name;
    private String description;
    private Boolean isPublic;
    private String address;
    private List<Screen> screens;
    private List<Client> clients;

    @Data
    @Builder
    public static class Client {
        private UUID uuid;
        private String name;
    }

    @Data
    @Builder
    public static class Screen {
        private UUID uuid;
        private String name;
    }
}

File: frontend/src/app/dtos/locationRequest.ts
import { LocationResourceResponse } from "./locationResourceResponse";

export class LocationResponse {
    constructor(
        public name: string | null = null,
        public description: string | null = null,
        public isPublic: boolean = true,
        public address: string | null = null,
        public screens: {
            uuid: string,
            name: string
        }[] = [],
        public clients: {
            uuid: string,
            name: string
        }[] = []
    ) {}
}
